src/Notifier.js

```diff
@@ -96,11 +96,52 @@ const Notifier = {
         }
     },
 
-    _playAudioNotification: function(ev, room) {
-        const e = document.getElementById("messageAudio");
-        if (e) {
-            e.play();
+    getSoundForRoom: async function(roomId) {
+        // We do no caching here because the SDK caches the event content
+        // and the browser will cache the sound.
+        let content = SettingsStore.getValue("notificationSound", roomId);
+        if (!content) {
+            content = SettingsStore.getValue("notificationSound");
+            if (!content) {
+                return null;
+            }
+        }
+
+        if (!content.url) {
+            console.warn(`${roomId} has custom notification sound event, but no url key`);
+            return null;
         }
+
+        return {
+            url: MatrixClientPeg.get().mxcUrlToHttp(content.url),
+            name: content.name,
+            type: content.type,
+            size: content.size,
+        };
+    },
+
+    _playAudioNotification: function(ev, room) {
+        this.getSoundForRoom(room.roomId).then((sound) => {
+            console.log(`Got sound ${sound.name || "default"} for ${room.roomId}`);
```

Comment:
This line causes an error when `sound` is undefined.
```suggestion
            console.log(`Got sound ${sound ? sound.name : "default"} for ${room.roomId}`);
```